src/frontend/src/components/public/PIANavButton/index.tsx

```diff
@@ -1,50 +1,92 @@
-import { useLocation, useNavigate } from 'react-router-dom';
-import { PiaFormSideNavPages } from '../PIASideNav/pia-form-sideNav-pages';
+import { Link, useLocation, useNavigate } from 'react-router-dom';
 import { INavButton } from './interface';
 
-const PIANavButton = ({ pia }: INavButton) => {
+const PIANavButton = ({ pages, isIntakeSubmitted, isDelegate }: INavButton) => {
   const { pathname } = useLocation();
   const navigate = useNavigate();
+
+  const handleNavBtn = (direction: string) => {
+    const current = pages.find((page) => page.link === pathname);
+    const currentIndex = pages.findIndex((page) => page.link === pathname);
+    if (current === undefined) return;
+    if (direction === 'next') {
+      if (current.state?.next?.condition) {
+        if (current.state?.next?.action === +1) {
+          return pages[currentIndex + 1].link;
+        } else if (typeof current.state?.next?.action === 'string') {
+          return pages.find(
+            (page) => page.label === current.state?.next?.action,
+          )?.link;
+        }
+      }
+    } else if (direction === 'back') {
+      if (current.state?.prev?.condition) {
+        if (current.state?.prev?.action === -1) {
+          return pages[currentIndex - 1].link;
+        } else if (typeof current.state?.prev?.action === 'string') {
+          return pages.find(
+            (page) => page.label === current.state?.prev?.action,
+          )?.link;
+        }
+      }
+    }
+  };
+
   const handleBack = () => {
-    navigate(-1);
+    const backLink = handleNavBtn('back');
+    if (backLink !== undefined) navigate(backLink);
   };
   const handleNext = () => {
-    const editMode = pathname.split('/')[4];
-    const pages = PiaFormSideNavPages(
-      pia,
-      editMode === 'edit' ? true : false,
-      false,
-    );
-    const currentTabObj = pages.filter((page) => page.link === pathname);
-    const nextTabObj = pages.filter(
-      (page) => page.id === currentTabObj[0].id + 1,
-    );
-    navigate(nextTabObj[0].link);
+    const nextLink = handleNavBtn('next');
+    if (nextLink !== undefined) navigate(nextLink);
   };
+
   return (
     <>
-      {!pathname.includes('intake') && (
-        <div>
-          <div className="horizontal-divider"></div>
-          <div className="form-buttons">
-            <button
-              className="bcgovbtn bcgovbtn__secondary btn-back"
-              onClick={handleBack}
-            >
-              Back
-            </button>
-            {!pathname.includes('additionalRisks') && (
-              <button
-                type="submit"
-                className="bcgovbtn  bcgovbtn__secondary btn-next"
-                onClick={handleNext}
-              >
-                Next
-              </button>
-            )}
-          </div>
-        </div>
-      )}
+      {!isDelegate
```

Comment:
This condition must come from the page->state>prev>condition. 
Why is there an if/else for this?

Reply:
the next Step page show the button differently based on PI or Non-PI. if it is PI, we show them next and back, but if non-PI, it display back and view pia list btn